src/sql/src/plan/statement/ddl.rs

```diff
@@ -783,16 +809,48 @@ pub fn plan_create_source(
         }
     };
 
+    // TODO(petrosagg): remove this inconsistency once INCLUDE (offset)
+    // syntax is implemented
+    let default_metadata = provide_default_metadata(&envelope, encoding.value_ref());
+    let metadata_columns = external_connector.metadata_columns(default_metadata);
     let (key_desc, value_desc) = encoding.desc()?;
-    let mut bare_desc = envelope.desc(key_desc.clone(), value_desc)?;
+    let metadata_desc = included_column_desc(metadata_columns.clone());
+    let mut bare_desc = envelope.desc(key_desc.clone(), value_desc, metadata_desc)?;
+
+    // Append default metadata columns if column aliases were provided but do not include them.
+    //
+    // This is a confusing hack due to two combined facts:
+    //
+    // * we used to not allow users to refer to/alias the metadata columns because they were
+    //   specified in render, instead of here in plan
+    // * we don't follow postgres semantics and allow a shorter rename list than total column list
+    //
+    // TODO: probably we should just migrate to pg semantics and allow specifying fewer columns than
+    // actually exist?
+    let tmp_col;
+    let col_names = if default_metadata
+        && !col_names.is_empty()
+        && metadata_columns.len() + col_names.len() == bare_desc.arity()
+    {
+        let mut tmp = Vec::with_capacity(bare_desc.arity());
+        tmp.extend(col_names.iter().cloned());
+        tmp.push(Ident::from(
+            external_connector.default_metadata_column_name().unwrap(),
+        ));
+        tmp_col = tmp;
+        &tmp_col
+    } else {
+        col_names
+    };
```

Comment:
This section replaces the RelationDesc stuff that used to happen in `metadata_columns()` 

https://github.com/MaterializeInc/materialize/pull/8935/files#diff-39dd885d0ba2f09d92b483c0504afd922b974405c91e5eaa7ef68e044af818ecL828-L833